eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/ChannelRelayer.scala

```diff
@@ -87,16 +88,16 @@ object ChannelRelayer {
               Behaviors.same
 
             case GetOutgoingChannels(replyTo, Relayer.GetOutgoingChannels(enabledOnly)) =>
-              val channels = if (enabledOnly) {
-                channelUpdates.values.filter(o => Announcements.isEnabled(o.channelUpdate.channelFlags))
-              } else {
-                channelUpdates.values
-              }
+              val channels = for {
+                out @ OutgoingChannel(_, channelUpdate, _: Commitments) <- channelUpdates.values // We are only interested in normal channel Commitments here
+                if !enabledOnly || (enabledOnly && Announcements.isEnabled(channelUpdate.channelFlags))
+              } yield out
+
```

Comment:
Why do you filter it out here? It feels very weird to filter here because of something external from eclair-core.
Is that filter really necessary? If so, can't it be done outside of here (in the plugins that enable HC)?

Reply:
It's done this way to preserve existing behavior, if HC is present in system then its commitments data will be injected into `ChannelRelayer` maps and will be returned alongside with normal channel commitments unless filtered. 

This may or may not be desired so I've decided to take a safe route and filter out anything except what core Eclair already understands.

Reply:
Would a user running a node with hosted channels be interested in getting them in the `channels` API?
It seems to me that yes, it would be interesting for them, wouldn't it?

I think it's weird to filter here for things eclair-core doesn't know about that will be built on top: if a user adds a plugin to eclair-core that adds hosted channels, that user knows about them and can decide whether to filter them or not, so the decision should be taken by the plugin?

The only reason I see for filtering them out explicitly is if adding hosted channels and having them returned here would break something that eclair-core does; you should check what actors depend on this channels list and evaluate whether that's the case.

Reply:
> Would a user running a node with hosted channels be interested in getting them in the channels API?
It seems to me that yes, it would be interesting for them, wouldn't it?

the plan for this in particular is to have a separate plugin-specific API (similar to keysend plugin which establishes its own API). 

But in general, yes, there is no reason to filter out HC in particular because they are designed to be indistinguishable from normal channels to the rest of the system. 

For instance, if user wants to send (or receive) a payment then not filtering HC out would make it a candidate for a first channel in a route when sending or another route hint when receiving, and it's perfectly fine. 

I guess this is a precautionary measure if someone decides to add some other type of injectable commitments via plugin without taking care of full compatibility, but maybe this is unjustified and this part should be well documented instead?

Reply:
Yes, I think that for now we shouldn't do any filtering.
It will be up to future plugin developers to figure out if they need their new channel type to be filtered out or not; if it needs to be filtered out, at that point a PR should be opened to eclair-core to introduce a filtering mechanism, or a mechanism in the plugin itself should be found.